pkg/spi-shared/config/defaults.go

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+// ServiceProviderDefaults configuration containing default values used to initialize supported service providers
+type ServiceProviderDefaults struct {
```

Comment:
what if we made `ServiceProviderType` a struct (with `Name` being the current value of the different types and additional fields as in this defaults struct)?